Adicionando 2557 (#198)
petbccufscar/URI: Strings/2557/README.md

``` c
            j=0;
            while (equacao[i] != 'R' && equacao[i] != 'L' && equacao[i] != 'J' && equacao[i] != '+' && equacao[i] != '=' && equacao[i] != '-' && i<tamanho) {
                valores[k][j] = equacao[i];
                if(equacao[i+1] == '+' || equacao[i+1] == '=' ||equacao[i+1] == '-' ) {
                    k++;
                }
                i++;
                j++;
            }
```

> Finalizado o loop anterior, tem-se que o primeiro membro da equação foi lido, sendo assim, torna-se necessário definir alguns parâmetros para uso futuro. Primeiramente, é necessário definir se o primeiro caractere lido foi um número, para usa-se um `if` e `i`, caso a variável ainda tenha seu valor zerado significa que foi lido uma letra, logo a váriavel de controle se o primeiro número está no lado esquerdo é zerada. Segundamente, confere-se a operação que será feita é uma soma ou divisão, para isso basta verificar se o caractere sendo acessado no momento por `i` é um sinal positivo, se sim, incrementa-se o valor de soma. Por fim, confere-se se alguma letra está do lado esquerdo da equação, para tal basta configura o acesso para ser limitado ao penultimo membro da equação e o primeiro número deve estar ao lado esquerdo. Ademais, incrementa-se `i` para acessar o próximo caractere. Assim finaliza-se o segundo loop.

Reviewer: ```suggestion
Finalizado o loop anterior, tem-se que o primeiro membro da equação foi lido, sendo assim, torna-se necessário definir alguns parâmetros para uso futuro. Primeiramente, é necessário definir se o primeiro caractere lido foi um número, para usa-se um `if`. Se `i` for 0, significa que foi lido uma letra, logo a  váriavel `lado_esquerdo1`  é zerada. Segundamente, confere se a operação que será feita é uma soma ou divisão, para isso basta  verificar se o caractere sendo acessado no momento por `i` é um sinal positivo, se sim, incrementa-se o valor de soma. Por fim, confere-se se alguma letra está do lado esquerdo da equação, para tal basta configurar o acesso para ser limitado ao penúltimo membro da equação (`i < tamanho-2`) e o primeiro número deve estar ao lado esquerdo (`lado_esquerdo1 == 1`). Ademais, incrementa-se `i` para acessar o próximo caractere. Assim finaliza-se o segundo loop.
```

``` c
if(i==0) {
                lado_esquerdo1 = 0;
            }
            else if(equacao[i] == '+') {
                soma++;
            }
            if((equacao[i+1] == 'R' || equacao[i+1] == 'L' ||equacao[i+1] == 'J') && lado_esquerdo1 == 1 && i<tamanho-2) {
                lado_esquerdo2 = 0;
            }
            i++;
        }
```

Avançando, usa-se a função `atoi`(resposável por transformar números de uma string em um dado do tipo `int`) para armazenar os números lidos na forma de números integrais.
``` c
        valor1 = atoi(valores[0]);
        valor2 = atoi(valores[1]);
```

> Dessa forma, têm-se o necessário para imprimir o resultado, basta estipular cada condição e sua operação respectiva - para tal usa-se estruturas condicionais alinhadas. A primeira condição é: a operação é uma soma e ambos os números estão do lado esquerdo, logo soma-se os valores; a segunda: a operação é uma soma, mas um dos número não está do lado direito da equação, logo subtrai-se do segundo valor o primeiro valor; a terceira: a operação é uma subtração e ambos os valores estão do lado esquerdo da igualdade, logo subtrai-se do primeiro valor o segundo valor; a quarta condição: a operação é uma subtração e um dos valoes está do lado direito da igualdade, logo subtrai-se do primeiro valor o segundo valor. 

Reviewer: ```suggestion
Dessa forma, têm-se o necessário para imprimir o resultado, basta estipular cada condição e sua operação respectiva - para tal usa-se estruturas condicionais alinhadas.
 * A primeira condição é: a operação é uma soma e ambos os números estão do lado esquerdo, logo soma-se os valores;
 * A segunda: a operação é uma soma, mas um dos número não está do lado direito da equação, logo subtrai-se do segundo valor o primeiro valor;
* A terceira: a operação é uma subtração e ambos os valores estão do lado esquerdo da igualdade, logo subtrai-se do primeiro valor o segundo valor; 
* A quarta condição: a operação é uma subtração e um dos valoes está do lado direito da igualdade, logo subtrai-se do primeiro valor o segundo valor.
```

``` c 
        if(soma == 1 && lado_esquerdo1==1 && lado_esquerdo2 == 1) {
            printf("%d\n", valor1 + valor2);
        }
        else if(soma == 1 && (lado_esquerdo1==0 || lado_esquerdo2 ==0)) {
            printf("%d\n", valor2 - valor1);
        }
        else if(soma == 0 && lado_esquerdo1 == 1 && lado_esquerdo2 == 1) {
            printf("%d\n", valor1 - valor2);
        }
        else if(soma==0 && (lado_esquerdo1==0 || lado_esquerdo2 ==0)){
            printf("%d\n", valor1 - valor2);
        }
```

Após a impressão do resultado, basta reiniciar os valores das variáveis para reiniciar o loop responsável pelos testes de caso. Assim sendo, configura as variáveis do tipo `int` para 0, e usa-se a função `memset` para zerar os valores das variáveis do tipo `char`. Dessa forma, o loop principal está encerrado e, para finalizar o programa, basta inserir `return 0`.
``` c
        soma = 0;
        i = 0;
        k = 0;
        j = 0;
        lado_esquerdo1 = 1;
        lado_esquerdo2 = 1;
        memset(valores[0], 0, 10);
        memset(valores[1], 0, 10);
    }
    return 0;
}
```

> ##### Para aprender um pouco mais sobre memset:[memset]http://www.cplusplus.com/reference/cstring/memset/

Reviewer: ```suggestion
##### Para aprender um pouco mais sobre memset:[memset](http://www.cplusplus.com/reference/cstring/memset/)
```

> ##### Para aprender um pouco mais sobre atoi:[atoi]https://www.cplusplus.com/reference/cstdlib/atoi/

Reviewer: ```suggestion
##### Para aprender um pouco mais sobre atoi:[atoi](https://www.cplusplus.com/reference/cstdlib/atoi/)
```
